errors.go

```diff
@@ -0,0 +1,248 @@
+package gobayeux
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	// ErrClientNotConnected is returned when the client is not connected
+	ErrClientNotConnected = errors.New("client not connected to server")
+
+	// ErrTooManyMessages is returned when there is more than one handshake message
+	ErrTooManyMessages = errors.New("more messages than expected in handshake response")
+
+	// ErrBadChannel is returned when the handshake response is on the wrong channel
+	ErrBadChannel = errors.New("handshake responses must come back via the /meta/handshake channel")
+
+	// ErrFailedToConnect is a general connection error
+	ErrFailedToConnect = errors.New("connect request was not successful")
+
+	// ErrNoSupportedConnectionTypes is returned when the client and server
+	// aren't able to agree on a connection type
+	ErrNoSupportedConnectionTypes = errors.New("no supported connection types provided")
+
+	// ErrNoVersion is returned when a version is not provided
+	ErrNoVersion = errors.New("no version specified")
+
+	// ErrMissingClientID is returned when the client id has not been set
+	ErrMissingClientID = errors.New("missing clientID value")
+
+	// ErrMissingConnectionType is returned when the connection type is unset
+	ErrMissingConnectionType = errors.New("missing connectionType value")
+)
+
+// ErrConnectionFailed is returned whenever Connect is called and it fails
+type ErrConnectionFailed struct {
```

Comment:
These should be named differently, e.g., `ConnectionFailedError`

Reply:
This naming convention comes from [staticcheck](https://staticcheck.io/docs/checks#ST1012)

Reply:
That's for variables, not types

Reply:
See also https://go.dev/doc/effective_go#errors